### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by byiringiroscar**
#### Decorate class
This PR is raised to implement the following features 🆗 :

- [x] Decorate class
- [x] Setup linters
- [x] Use modules

### How should it be tested ⬇️ ⬇️ ⬇️ 

clone this repository
     
```
git clone https://github.com/byiringiroscar/OOP-school-library-with-Ruby/
```
Navigate to folder
```
cd OOP-school-library-with-Ruby
```
Switch to this feature/decorate
```
git switch feature/decorate
```
Run project and Installation
- open the command prompt and run :
```
ruby test.rb
```

### Thanks for reviewing my PR 😄

### CODE REVIEW

File: base.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (nahnahdev) [inline base.rb]: - [ ] Great job so far @byiringiroscar However, based on ruby conventions each class should be in its own file. Kindly move this class to its own file, then import it instead, for example:

`require_relative 'capitalize.rb`

File: base.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    trimmed_name = @nameable.correct_name
+    trimmed_name.length > 10 ? trimmed_name[0..9] : trimmed_name
+  end
+end
```
Comment (nahnahdev) [inline base.rb]: - [ ] Just as I mentioned before, based on ruby conventions each class should be in its own file. Kindly move this class to its own file, then import it instead 👍

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @byiringiroscar , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- App works as expected 🔥
- Professional README 🔥
- No linter errors 🔥

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by levy002 — APPROVED** [review]
## Hello @byiringiroscar 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

  unnamed


## Highlights
- No linter error :heavy_check_mark: 
- You implemented all suggested changes :heavy_check_mark: 
 
## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**